airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -84,3 +84,73 @@ async def run(self):
                     raise AirflowException(f"Dataproc job execution failed {self.job_id}")
             await asyncio.sleep(self.polling_interval_seconds)
         yield TriggerEvent({"job_id": self.job_id, "job_state": state})
+
+
+class DataprocWorkflowTrigger(BaseTrigger):
```

Comment:
does it make sense to extend this class from `DataprocBaseTrigger` instead of  `BaseTrigger` https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/triggers/dataproc.py#L31 WDYT?

Reply:
Yeah. Would it?

Reply:
DataprocBaseTrigger is dedicated to e.g. DataprocSubmitJobOperator.
I changed the name to DataprocSubmitTrigger for clarity.

Reply:
@VladaZakharova  @bkossakowska somehow I thought this PR has merged so I created https://github.com/apache/airflow/pull/29364 PR to do some refactoring on the trigger and address my above comments.

Reply:
merging anyone would cause conflict in others.

Reply:
@pankajastro 
Okay, no problem!
Do you know when your changes for the trigger will be merged?